app/services/merge_patient_service.rb

```diff
@@ -4,6 +4,7 @@ def initialize(payload)
   end
 
   def merge
+    existing_patient_attributes = Patient.with_discarded.find_by(id: payload['id'])&.attributes
```

Comment:
Yes. Since the patient payload contains patients as well as their address, phone numbers etc., this service handles pushing updates to the individual models